### ISSUE TITLE
[Blutopia] Some Search Issues

### ISSUE THREAD

**Issue by HDVinnie** [Jackett/Jackett#6854]
For instance, Jackett currently finds no torrents for:

`game of thrones S01 remux`

Searching `game of thrones S01 remux` directly on Blutopia website however returns two results. 
```
Game of Thrones S01 2160p UHD BluRay REMUX HDR HEVC Atmos-EPSiLON
Game of Thrones S01 1080p Hybrid Remux AVC TrueHD 7.1-BiZKiT
```

Searching just "game of thrones" in Jackett nets the expected amount of results (76 specifically) in Jackett which matches Blutopia.

**Comment by garfield69** [comment]
I don't have an a/c on blutopia so I cannot help with this one.
looks like its invitations only, because I'm not going to pass all the `i'm an enthusiastic sharer` hoop juimping for the regular registration.

If you send an invite to garfieldsixtynine @ gmail.com I can take a look, otherwise it will have to wait for another volunteer to investigate.

**Comment by ngosang** [comment]
Tests:
* Remove `$raw: "{{ range .Categories }}categories[]={{.}}&{{end}}"` => works but no category filter
*  Test in postman, send field `search=game%20of%20thrones%20S01%20remux&categories[]=2` =>  works
*  Test in postman, send field `search=game%20of%20thrones%20S01%20remux&categories[]=2,3` =>  works

* Test in postman, send field `search=game%20of%20thrones%20S01%20remux&categories[]=` => **don't work**

**Comment by HDVinnie** [comment]
@garfield69 I have sent a invite. I should also note Blutopia does offer a API for searching torrents. Not sure if could be of use.

https://blutopia.xyz/pages/19

**Comment by garfield69** [comment]
registration and login successful. thank you.

**Comment by garfield69** [comment]
its the use of a capital **S01** in the search request.

But its a case of a perfect storm.
# TLDR;
combo of 
(a) Blutopia name search season position, which works when S08 is in the middle of the name search but not when it is at the end
along with 
(b) Jackett ?Cardigann? changing the keyword order when capital S08 is used (shifting it to the end of the keyword list),  but not when lowercase s08 (which remains in place).

# more details:
1. it appears that Blutopia **name** search is positional depended, when it comes to season in the name.
for example, on the Blutopia website, searching **name** `game of thrones S08 1080p remux` yields 1 result `Game of Thrones S08 1080p BluRay REMUX AVC Atmos-EPSiLON` 
but searching for **name** `game of thrones 1080p remux S08` yields no results.

2.  the URL generated by cardigann is different when lowercase `s08` is used as opposed to uppercase `S08`
(a) Jackett search for lowercase `game of thrones s08 1080p remux` returns expected result, with the URL generated being `https://blutopia.xyz/torrents/filter?search=game+of+thrones+s08+1080p+remux&description=&uploader=&imdb=&tvdb=&tmdb=&mal=&igdb=&sorting=created_at&sort=created_at&direction=desc&qty=100`

(b) Jackett search for uppercase `game of thrones S08 1080p remux` returns nothing, the  URL generated is  `https://blutopia.xyz/torrents/filter?search=game+of+thrones++1080p+remux+S08&description=&uploader=&imdb=&tvdb=&tmdb=&mal=&igdb=&sorting=created_at&sort=created_at&direction=desc&qty=100`

note the uppercase `S08` in the URL has been shifted to the end of the **search=** parm. WTF?

Is Cardigann changing the order of the keywords? I don't know, that is a question  for a C# dev to examine and answer. Beyond my skills to analyse.

more investigation required. C# dev invited.

**Comment by RoloSoze** [comment]
First off, if someone needs access to Blutopia, their application signup is currently open:
https://blutopia.xyz/application

Secondly, @garfield69 BeyondHD has a very similar issue. If you enter the movie/TV show's name, it will return the correct results. Once you enter things like resolution, release group, REMUX, etc, it'll return nothing. 3 things worth mentioning here:

1. This happens sometimes, not always. I can't figure out why.
2. Sometimes this also happens on the BHD's website
3, We've had this issue with BHD in the past
https://github.com/Jackett/Jackett/issues/835
https://github.com/Jackett/Jackett/issues/4828

However, back then BHD ran on a different platform and had a completely different design.

**Comment by HDVinnie** [comment]
This is a problem with UNIT3D. I will be refactoring the search engine. As for BHD they are using a fork of UNIT3D they have modified. I just tested and they indeed still have same issue . So I suggest you will have to contact them regarding their site seeing they dont keep up to date with UNIT3D core. Im going to be starting the new search system this weekend so all UNIT3D sites that keep up to date with UNIT3D should be good soon.

**Comment by RoloSoze** [comment]
@HDVinnie I'm facing a similar issue. If I try to search for Cheers S01 1080p, Blutopia doesn't return anything in Jackett but does return 2 results when searching the website. 

However, if I remove 1080p, Jackett returns the 2 relevant results.

**Comment by RoloSoze** [comment]
Any progress on this issue, @HDVinnie?

**Comment by galfwender** [comment]
Hello,

Similar experiences here but I am not sure if this is related to the above.  Blutopia had been working fine for me but currently it spams with lots of unwanted results every time I search using Sonarr.

As an example, in the Jackett web interface if I search for "Blackadder S01", it returns a correct result from Blutopia - just a single dvd remux for that season.  Doing the interactive season search via Sonarr shows the following query in Jackett logs:

Fetching: https://blutopia.xyz/torrents/filter?categories[]=2&categories[]=6&search=S01&description=&uploader=&imdb=&tvdb=76736&tmdb=&mal=&igdb=&sorting=created_at&sort=created_at&direction=desc&qty=100

It is searching using TVDB id, and name of "S01".  The result is Sonarr lists many many results, anything matching "S01" as this param overrides the TVDB id in Blutopia.  Using just the TVDB id by itself shows only results related to Blackadder, which would be more preferable, but I have no way to make this happen.

**Comment by HDVinnie** [comment]
> Hello,
> 
> Similar experiences here but I am not sure if this is related to the above. Blutopia had been working fine for me but currently it spams with lots of unwanted results every time I search using Sonarr.
> 
> As an example, in the Jackett web interface if I search for "Blackadder S01", it returns a correct result from Blutopia - just a single dvd remux for that season. Doing the interactive season search via Sonarr shows the following query in Jackett logs:
> 
> Fetching: [https://blutopia.xyz/torrents/filter?categories[]=2&categories[]=6&search=S01&description=&uploader=&imdb=&tvdb=76736&tmdb=&mal=&igdb=&sorting=created_at&sort=created_at&direction=desc&qty=100](https://blutopia.xyz/torrents/filter?categories%5B%5D=2&categories%5B%5D=6&search=S01&description=&uploader=&imdb=&tvdb=76736&tmdb=&mal=&igdb=&sorting=created_at&sort=created_at&direction=desc&qty=100)
> 
> It is searching using TVDB id, and name of "S01". The result is Sonarr lists many many results, anything matching "S01" as this param overrides the TVDB id in Blutopia. Using just the TVDB id by itself shows only results related to Blackadder, which would be more preferable, but I have no way to make this happen.

TMDB IDs are primary on BLU with IMDB IDs being secondary. I would not rely on searching by TVDB IDs. However why it is removing Blackadder from search is beyond me. UNIT3D still needs a large search system refactor which I know I said I was starting awhile back. However it was halted due to more important issues and looking into new search technologies.

**Comment by ilike2burnthing** [comment]
@HDVinnie given a lot has changed since 2020, are any of these issues still outstanding?

**Comment by HDVinnie** [comment]
Not that im aware of.

BLU API also now supports passing seasonNumber and episodeNumber parameters. For example:

https://blutopia.xyz/api/torrents/filter?api_token=xxx&imdbId=0182576&seasonNumber=20&episodeNumber=10&sortField=created_at&sortDirection=desc&perPage=100&page=1&categories[]=2

or

https://blutopia.xyz/api/torrents/filter?api_token=xx&name=Family%20Guy&seasonNumber=20&episodeNumber=10&sortField=created_at&sortDirection=desc&perPage=100&page=1&categories%5B%5D=2

**Comment by ilike2burnthing** [comment]
Thanks.

Yes, implementing that is being looked into here - https://github.com/Prowlarr/Indexers/pull/96
